root/components/forms.tt

```diff
@@ -187,18 +187,48 @@
     [% END %]
 [%- END -%]
 
+[%- MACRO too_short_year_error(id) BLOCK -%]
+  <div class="error field-error" id="[% id %]" style="display: none">
+    [% WRAPPER form_row %]
+      [%- l('The year should have four digits. If you want to enter a year earlier than 1000 CE, please pad with zeros, such as “0123”.') -%]
+    [% END %]    
+  </div>
+[%- END -%]
+
 [%- MACRO date_range_fieldset(r, type, ended_label) BLOCK -%]
     <fieldset>
       <legend>[% l('Date Period') %]</legend>
       <p>
         [%- l('Dates are in the format YYYY-MM-DD. Partial dates such as YYYY-MM or just YYYY are OK, or you can omit the date entirely.') -%]
       </p>
       [% form_row_date(r, 'period.begin_date', l('Begin date:')) %]
+      [% too_short_year_error('too_short_begin_year') %]
       [% form_row_date(r, 'period.end_date', l('End date:')) %]
+      [% too_short_year_error('too_short_end_year') %]
       [% form_row_checkbox(r, 'period.ended', ended_label) %]
     </fieldset>
     <script type="text/javascript">
       MB.initializeToggleEnded('id-edit-[% type %]');
+
+      function blockTooShortBeginYear() {
+        var beginYear = $('#id-edit-[% type %]\\\.period\\\.begin_date\\\.year').val();
+        var allowed = (beginYear === null || beginYear === '' || beginYear.length === 4);
```

Comment:
I don't think the value can be `null` there, only `undefined` if the input doesn't exist (but it should exist). A slightly more robust check might be `!beginYear || beginYear.trim().length === 4`, which also removes whitespace before checking the length.